src/source/query_features.js

```diff
@@ -54,9 +62,18 @@ export function queryRenderedFeatures(sourceCache: SourceCache,
     // Merge state from SourceCache into the results
     for (const layerID in result) {
         result[layerID].forEach((featureWrapper) => {
-            const feature = featureWrapper.feature;
+            const feature: GeoJSONFeature & {
+                state: FeatureStates;
+                source: string;
+                sourceLayer: string;
+                layer: LayerSpecification;
+            } = (featureWrapper.feature: any);
+
+            // $FlowFixMe prop-missing
```

Comment:
`background` and `sky` layers don't have a source, but skipping this step breaks `test-query` suite, ignoring for now